src/net/sourceforge/kolmafia/KoLCharacter.java

```diff
@@ -2326,6 +2327,32 @@ public static final boolean hasGs(String name) {
     return name.contains("g") || name.contains("G");
   }
 
+  private static final Pattern I_PATTERN = Pattern.compile("[Ii]");
+
+  public static final int getLetterIs(String name) {
+    return (int) KoLCharacter.I_PATTERN.matcher(name).results().count();
+  }
+
+  public static final boolean hasLetterIs(String name) {
+    if (name == null) {
+      return false;
+    }
+    return name.contains("i") || name.contains("I");
+  }
```

Comment:
This could call your method which counts letter I - iterating once over the characters in the string - rather than 2 calls to contains.